File: 645-set-mismatch/set-mismatch.cpp
class Solution {
    public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int>answer(2);
        int n=nums.size();
        sort(nums.begin(), nums.end());
        for(int i=0; i<n-1; i++){
            if(nums[i]==nums[i+1]){
                answer[0]=nums[i];
                break;
            }
        }
        int sm=accumulate(nums.begin(),nums.end(),0);
        int res = n*(n+1)/2 - sm + answer[0];
        answer[1]=res;
        return answer;
    }
};

File: 645-set-mismatch/set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        answer = []
        nums.sort()
        n = len(nums)
        dup = 0
        array_sum = sum(nums)
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                answer.append(nums[i])
                dup = nums[i]

        total_sum = (n*(n+1)/2) - array_sum + dup
        answer.append(total_sum)
        return answer
